contents/security/sqlInjection.md

```diff
@@ -12,37 +12,71 @@
+
+## Why is it Important to Prevent SQL Injection
+1. **SQL injection is frequently exhibited.** According to [Open Web Application Security Project(OWASP) report](https://www.owasp.org/images/7/72/OWASP_Top_10-2017_%28en%29.pdf.pdf), the injection attack is always the annual top 1 application security risk from 2013 to 2017. In addition, [Statistics from Akamai](https://www.akamai.com/uk/en/resources/our-thinking/state-of-the-internet-report/web-attack-visualization.jsp) shows that over 80% of attacks are SQL injection in one week.
+1. **SQL injection can cause serious consequences** On one side, SQL injection can cause the loss of large amount of money. Based on the [Global Threat Intelligence Report](https://www.helpnetsecurity.com/2014/03/28/analysis-of-three-billion-attacks-reveals-sql-injections-cost-196000/), even a small SQL injection attack may cause hundreds of thousands dollars lost. 
+On the other side, information disclosure is another serious consequence. [An SQL injection attack to a toymaker company VTech](https://coar.risc.anl.gov/consequences-of-sql-injection-attacks/) has caused millions of parents' and children's profiles stolen. 
+Thus, we need to pay attention to prevent this attack in our code.
+
+## How does SQL Injection Work
+
+### CIA: Confidentiality, Integratity, Availability
+Before looking through SQL injection, we first look at three main requirements of computer security: confidentiality, integratity, and availability. We can understand them intuitively as follow:
+* **Confidentiality**: unauthorized people cannot view the information.
+* **Integratity**: unauthorized people cannot modifiy the information.
+* **Availability**: unauthorized people cannot prevent authorized people accessing the information.
+
+It turns out that SQL injection can compromise all these three security requirements. We will illustrate this through two senarios: inject more parameters to a query and inject more queries.
+
+### Inject extra parameters to compromise authentication
+Websites combine SQL syntax with user input to query the underlying database of the website. For example, most websites may ask for a username and password from users for them to login to the site. This input is used to create the SQL statement that will be sent to the database and verify the user's identity. 
+
+However, a hacker may utilize the SQL injection to login other user's account and access his account information. This compromise confidentiality. Moreover, if the user has privilege to modify sensitive information, (e.g. a lecture who can modify students' marks) it will break integrity as well.
+
+To illustrate how does SQL injection work, we first look at the workflow of the website.
```

Comment:
Maybe "To illustrate how SQL injection works, let's look at an example workflow of a website." sounds better?